File: app/src/main/java/com/nearsoft/nearbooks/gcm/NearbooksInstanceIDListenerService.java
package com.nearsoft.nearbooks.gcm;

import android.content.Intent;

import com.google.android.gms.iid.InstanceIDListenerService;

/**
 * Instance id listener.
 * Created by epool on 1/19/16.
 */
public class NearbooksInstanceIDListenerService extends InstanceIDListenerService {

    @Override
    public void onTokenRefresh() {
        Intent intent = new Intent(this, NearbooksRegistrationIntentService.class);
        startService(intent);
    }

}

File: app/src/main/java/com/nearsoft/nearbooks/gcm/NearbooksInstanceIDListenerService.kt
package com.nearsoft.nearbooks.gcm

import android.content.Intent

import com.google.android.gms.iid.InstanceIDListenerService

/**
 * Instance id listener.
 * Created by epool on 1/19/16.
 */
class NearbooksInstanceIDListenerService : InstanceIDListenerService() {

    override fun onTokenRefresh() {
        val intent = Intent(this, NearbooksRegistrationIntentService::class.java)
        startService(intent)
    }

}
